File: sdk/php/src/Resources/Ipam.php
<?php

namespace HiTechCloud\SDK\Resources;

class Ipam extends BaseResource
{
    /**
     * List IPs
     */
    public function listIps(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/htcipam/ips');
    }

    /**
     * List subnets
     */
    public function listSubnets(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/htcipam/subnets');
    }

    /**
     * List rDNS
     */
    public function listRdns(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/htcipam/rdns');
    }

    /**
     * Set rDNS
     */
    public function setRdns(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/htcipam/rdns', $data);
    }

}

File: sdk/csharp/Resources/Ipam.cs
namespace HiTechCloud.SDK.Resources;

public class Ipam : BaseResource
{
    public Ipam(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> IpsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/htcipam/ips", id));
    }

    public async Task<Dictionary<string, object?>> SubnetsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/htcipam/subnets", id));
    }

    public async Task<Dictionary<string, object?>> RdnsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/htcipam/rdns", id));
    }

    public async Task<Dictionary<string, object?>> SetRdnsAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/htcipam/rdns", id), data);
    }
}
